week-3/slideshow/slideshow.js

```diff
@@ -1,8 +1,131 @@
 const images = [
-    "./assets/cute-cat-a.png",
-    "./assets/cute-cat-b.jpg",
-    "./assets/cute-cat-c.jpg",
+  "./assets/cute-cat-a.png",
+  "./assets/cute-cat-b.jpg",
+  "./assets/cute-cat-c.jpg",
 ];
 
+let currentPhotoIndex = 0;
 
-// Write your code here
\ No newline at end of file
+const image = document.getElementById("carousel-img");
+
+const imgContainer = document.createElement("div");
+imgContainer.setAttribute("id", "img-container");
+
+image.before(imgContainer);
+imgContainer.appendChild(image);
+
+const btnContainer = document.createElement("div");
+btnContainer.classList.add("btn-container");
+
+const backwardBtn = document.getElementById("backward-btn");
+const forwardBtn = document.getElementById("forward-btn");
+
+const stopBtn = document.getElementById("stop");
+
+const autoBackwardBtn = document.getElementById("auto-backward");
+const autoForwardBtn = document.getElementById("auto-forward");
+
+btnContainer.append(
+  autoBackwardBtn,
+  backwardBtn,
+  stopBtn,
+  forwardBtn,
+  autoForwardBtn
+);
+
+let autoPlayBackward;
+let autoPlayForward;
+
+// Write your code here
+
+function generatePhoto() {
+  image.setAttribute("src", images[currentPhotoIndex]);
+  if (currentPhotoIndex === 0) {
+    document.getElementById("img0").style.backgroundColor = "grey";
+    document.getElementById("img1").style.backgroundColor = "black";
+    document.getElementById("img2").style.backgroundColor = "black";
+  } else if (currentPhotoIndex === 1) {
+    document.getElementById("img0").style.backgroundColor = "black";
+    document.getElementById("img1").style.backgroundColor = "grey";
+    document.getElementById("img2").style.backgroundColor = "black";
+  } else if (currentPhotoIndex === 2) {
+    document.getElementById("img0").style.backgroundColor = "black";
+    document.getElementById("img1").style.backgroundColor = "black";
+    document.getElementById("img2").style.backgroundColor = "grey";
+  }
+}
+
+function nextPhoto() {
+  ++currentPhotoIndex;
+  if (currentPhotoIndex === 3) {
+    currentPhotoIndex = 0;
+  }
+  generatePhoto();
+}
+
+function previousPhoto() {
+  --currentPhotoIndex;
+  if (currentPhotoIndex === -1) {
+    currentPhotoIndex = 2;
+  }
+  generatePhoto();
+}
+
+backwardBtn.addEventListener("click", previousPhoto);
+forwardBtn.addEventListener("click", nextPhoto);
+
+autoBackwardBtn.addEventListener("click", () => {
+  autoPlayBackward = setInterval(() => {
+    previousPhoto();
+  }, 2000);
+  clearInterval(autoPlayForward);
```

Comment:
I love this!  it clears other intervals when you change the slide direction - such a thoughtful touch! Well done!